rescript/filter_length.py

```diff
@@ -96,6 +97,47 @@ def filter_seqs_length_by_taxon(sequences: DNAFASTAFormat,
     return result, failures
 
 
+def filter_taxa(taxonomy: pd.Series, ids_to_keep: qiime2.Metadata = None,
+                include: str = None, exclude: str = None) -> pd.Series:
+    if include is exclude is ids_to_keep is None:
+        raise ValueError('No filtering criteria were applied!')
+
+    ids = taxonomy.index
+    print('Input features: ' + str(len(ids)))
+    # some initial validation
+    if ids_to_keep:
+        ids_to_keep = ids_to_keep.ids
+        _index_is_superset(set(ids_to_keep), ids)
+
+    if include:
+        include = '|'.join(include)
+        ids = ids[taxonomy.str.contains(include.replace(',', '|'))]
```

Comment:
Took me a while to figure out what was going on here. I suppose this combination is to protect against whether or not a user is placing spaces and/or commas in between the taxonomies they wish to filter? That is either "Bacteria Archaea" or "Bacteria,Archaea" will work here. At leas that is what seems to be the case playing around in ipython. I like it.

Reply:
oh thanks for catching! silly oversight on my part, actually... I originally made `include` a `Str`, then changed to a `List[Str]`, so the `replace` is extraneous here. I removed that part.